File: Server/src/main/java/core/game/content/dialogue/TostigDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Handles the TostigDialogue dialogue.
 * @author 'Vexia
 */
@Initializable
public class TostigDialogue extends DialoguePlugin {

	public TostigDialogue() {

	}

	public TostigDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {

		return new TostigDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Hi, what ales are you serving?");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Well sir, our speciality is Asgarnian Ale, we also serve", "Wizard's Mind Bomb and Dwarven Stout.");
			stage = 1;
			break;
		case 1:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Would you like to buy a drink?");
			stage = 2;
			break;
		case 2:
			interpreter.sendOptions("Select An Option", "Yes, please.", "No, thanks.");
			stage = 3;
			break;
		case 3:
			switch (buttonId) {
			case 1:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Yes, please.");
				stage = 100;
				break;
			case 2:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "No, thanks.");
				stage = 4;
				break;
			}
			break;
		case 4:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Ah well... so um... does the grey squirrel sing in the", "grove?");
			stage = 5;
			break;
		case 5:
			interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Huh?");
			stage = 6;
			break;
		case 6:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Er... nevermind.");
			stage = 7;
			break;
		case 7:
			end();
			break;
		case 100:
			end();
			npc.openShop(player);
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 1079 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/dialogue/TostigDialogue.kt
package content.region.asgarnia.burthorpe.dialogue

import core.api.openNpcShop
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.dialogue.Topic
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.END_DIALOGUE
import core.tools.START_DIALOGUE
import org.rs09.consts.NPCs

/**
 * Handles the TostigDialogue dialogue.
 * @author 'Vexia
 */
@Initializable
class TostigDialogue(player: Player? = null) : DialoguePlugin(player) {
    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        when (stage) {
            0 -> playerl(FacialExpression.FRIENDLY, "Hi, what ales are you serving?").also { stage++ }
            1 -> npcl(FacialExpression.FRIENDLY, "Well ${if (player.isMale) "mister" else "miss"}, our speciality is Asgarnian Ale, we also serve Wizard's Mind Bomb and Dwarven Stout.").also { stage++ }
            2 -> npcl(FacialExpression.FRIENDLY, "Would you like to buy a drink?").also { stage++ }
            3 -> showTopics(
                Topic(FacialExpression.FRIENDLY, "Yes, please.", 10),
                Topic(FacialExpression.FRIENDLY, "No, thanks.", 20),
            )
            10 -> openNpcShop(player, npc.id).also {
                end()
                stage = END_DIALOGUE
            }
            20 -> npcl(FacialExpression.FRIENDLY, "Ah well... so um... does the grey squirrel sing in the grove?").also { stage++ }
            21 -> showTopics(
                Topic(FacialExpression.FRIENDLY, "Huh?", 30),
                Topic(FacialExpression.FRIENDLY, "Yes, and the black cat dances in the sink.", 40),
                Topic(FacialExpression.FRIENDLY, "No, squirrels can't sing.", 50),
            )
            30 -> npcl(FacialExpression.FRIENDLY, "Er... nevermind.").also { stage = END_DIALOGUE }
            40 -> npcl(FacialExpression.FRIENDLY, "Ah you'll be wanting the trapdoor behind the bar.").also { stage = END_DIALOGUE }
            50 -> npcl(FacialExpression.FRIENDLY, "No... of course they can't...").also { stage = END_DIALOGUE }
        }
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.TOSTIG_1079)
    }
}
